src/vivarium/framework/results/context.py

```diff
@@ -0,0 +1,124 @@
+from collections import defaultdict
```

Comment:
You've actually added a lot more in this PR than I was expecting. I had envisioned this ticket to be no more than renaming a few methods, updating documentation, and verifying that the existing tests for creating the interface and manager succeed. 

This is ok, but now you've introduced some real functionality into the codebase without the corresponding tests.

In `ResultsContext`, `set_default_stratifications`, `add_observation`, and `_format_results` are fully implemented.

From what I can tell, you've fully implemented `ResultsManager` which I envisioned being broken up across MIC-3133, MIC-3134, MIC-3341, and MIC-3349

I'll leave it up to you whether you want to remove these things or leave them in, but the benefits to adding them incrementally is that it will be much clearer what has been fully implemented inclusive of unit/integration tests. I think it will also be easier for people to follow the PRs.

Reply:
I've added a TODO for the ResultsContext testing.